src/traces/volume4d/convert.js

```diff
@@ -0,0 +1,132 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var createMesh = require('gl-mesh3d');
+
+var parseColorScale = require('../../lib/gl_format_color').parseColorScale;
+var str2RgbaArray = require('../../lib/str2rgbarray');
+var zip3 = require('../../plots/gl3d/zip3');
+
+var findNearestOnAxis = require('../isosurface/fn_find-nearest-on-axis');
+var generateIsoMeshes = require('../isosurface/fn_generate-iso-meshes');
+
+function Volume4dTrace(scene, mesh, uid) {
+    this.scene = scene;
+    this.uid = uid;
+    this.mesh = mesh;
+    this.name = '';
+    this.data = null;
+    this.showContour = false;
+}
+
+var proto = Volume4dTrace.prototype;
+
+proto.handlePick = function(selection) {
+    if(selection.object === this.mesh) {
+
+        var rawId = selection.data.index;
+
+        var x = this.data._x[rawId];
+        var y = this.data._y[rawId];
+        var z = this.data._z[rawId];
+
+        var height = this.data._Ys.length;
+        var depth = this.data._Zs.length;
+
+        var i = findNearestOnAxis(x, this.data._Xs).id;
```

Comment:
... and "nearest" here means nearest on the "top" slice, correct?

Reply:
... which I think is _ok_ for the first iteration, but could be potentially improved in the future.

Reply:
Would you mind hovering on `gl3d_volume4d_airflow` in your browser?